File: padroesgof/src/main/java/com/padroesestruturais/Adapter/PagXPTO.java
package com.padroesestruturais.Adapter;

/**
 * Broker de Pagamento da XPTO
 */
public class PagXPTO {
    
    public void pagarCartaoVisa(CartaoCredito cartao, double valor) {
        if(cartao.abaixarLimite(valor)) {
            System.out.println("XPTO: Pagamento de "+valor+" CONFIRMADO para cartão Visa do titular: " + cartao.getNome());
        } else {
            System.out.println("XPTO: Pagamento de "+valor+" NÃO CONFIRMADO para cartão Visa do titular: " + cartao.getNome());
        }
    }

    public void pagarCartaoMastercard(CartaoCredito cartao, double valor) {
        if(cartao.abaixarLimite(valor)) {
            System.out.println("XPTO: Pagamento de "+valor+" CONFIRMADO para cartão Mastercard do titular: " + cartao.getNome());
        } else {
            System.out.println("XPTO: Pagamento de "+valor+" NÃO CONFIRMADO para cartão Mastercard do titular: " + cartao.getNome());
        }
    }

}

File: PadroesGofCSharp/FormasDePagamento/PagXPTO.cs
using PadroesGofCSharp.Models;

namespace PadroesGofCSharp.FormasDePagamento;

public class PagXPTO
{
    public void pagarCartaoVisa(CartaoCredito cartao, decimal valor) {
        if(cartao.abaixarLimite(valor)) {
            Console.WriteLine("XPTO: Pagamento de "+valor+" CONFIRMADO para cartão Visa do titular: " + cartao.getNome());
        } else {
            Console.WriteLine("XPTO: Pagamento de "+valor+" NÃO CONFIRMADO para cartão Visa do titular: " + cartao.getNome());
        }
    }

    public void pagarCartaoMastercard(CartaoCredito cartao, decimal valor) {
        if(cartao.abaixarLimite(valor)) {
            Console.WriteLine("XPTO: Pagamento de "+valor+" CONFIRMADO para cartão Mastercard do titular: " + cartao.getNome());
        } else {
            Console.WriteLine("XPTO: Pagamento de "+valor+" NÃO CONFIRMADO para cartão Mastercard do titular: " + cartao.getNome());
        }
    }
}